cms/server/contest/handlers/base.py

```diff
@@ -58,11 +59,25 @@ class BaseHandler(CommonRequestHandler):
 
     def __init__(self, *args, **kwargs):
         super(BaseHandler, self).__init__(*args, **kwargs)
-        self.all_translations = None
+        # The list of interface translations the user can choose from.
+        self.available_translations = self.service.translations
+        # The locale that best matches the user's system settings (as
+        # reflected by the browser in the HTTP request's
+        # Accept-Language header).
+        self.browser_lang = DEFAULT_TRANSLATION.identifier
+        # The locale that the user specifically manually picked.
         self.cookie_lang = None
-        self.browser_lang = None
-        self.translation = None
-        self._ = None
+        # The translation that we are going to use.
+        self.translation = DEFAULT_TRANSLATION
+        self._ = self.translation.gettext
+
+        # We need this to be computed for each request because we want to be
+        # able to import new contests without having to restart CWS. But only
+        # in multi-contest mode.
+        self.contest_list = {}
+        if self.is_multi_contest():
+            for contest in self.sql_session.query(Contest).all():
+                self.contest_list[contest.name] = contest
```

Comment:
I can do it later, since it was already like this: this is only needed in the get of this handler, not for all handlers.

Reply:
I agree that there's no reason to get *all* contests in *every* request, since they are needed only when showing the contest list. Though be aware that `ContestHandler.choose_contest` uses this dict to pick the contest.